src/document.hpp

```diff
@@ -40,6 +40,13 @@ struct PageOffsets {
     int32_t page_obj_num;
 };
 
+struct PageLabel {
+    PageId page_id;
+    CapyPDF_PageLabelNumberStyle style;
+    u8string prefix;
+    int32_t start_num;
```

Comment:
According to the spec all of these are optional so it should be `std::optional`s (the string can be empty).

This should check if the page number points to outside defined pages and error out if it does.

As for the PageID. I'm not so sure if adding it was such a great idea. There are several places where you want to point to physical pages in the public API and creating PageID structs for that is a bit cumbersome. But that's a design issue for a different day.

Reply:
Yeah, technically we could have two different things; one being the label attached to the actual page object which we can construct at serialisation time and the other is the label schemes, which applies to multiple pages even though it specifies only the first page.